gef.py

```diff
@@ -8151,6 +8160,100 @@ def clean(self, event):
         return
 
 
+@register_command
+class IsSyscallCommand(GenericCommand):
+    """
+    Checks if the current instruction is a system call."""
+    _cmdline_ = 'is-syscall'
+    _syntax_ = _cmdline_
+
+    def do_invoke(self, argv):
+        insn = gef_current_instruction(current_arch.pc)
+        ok('Current instruction is{}a syscall'.format(' ' if self.is_syscall(current_arch, insn) else ' not '))
+
+        return
+
+    def is_syscall(self, arch, instruction):
+        insn_str = instruction.mnemonic  + ' ' + ', '.join(instruction.operands)
+        return insn_str.strip() in arch.syscall_instructions
+
+
+@register_command
+class SyscallArgsCommand(GenericCommand):
+    """
+    Gets the syscall name and arguments based on the register values in the current state."""
+    _cmdline_ = 'syscall-args'
+    _syntax_ = _cmdline_
+
+    def __init__(self):
+        super(SyscallArgsCommand, self).__init__()
+        self.add_setting("path", os.path.join(GEF_TEMP_DIR, "syscall-tables"),
+                         "Path to store/load the syscall tables files")
+        return
+
+    def do_invoke(self, argv):
+        color = get_gef_setting("theme.xinfo_title_message")
+
+        path = self.get_settings_path()
+        if path is None:
+            err("Cannot open '{0}': check directory and/or `gef config {0}` setting, currently: '{1}'".format("syscall-args.path",
+                                                                                                        self.get_setting("path")))
+            return
+
+        arch = current_arch.__class__.__name__
+
+        syscall_table = self.get_syscall_table(arch)
+        syscall_entry = syscall_table[get_register(current_arch.syscall_register)]
+
+        values = []
+        for param in syscall_entry.params:
+            values.append(get_register(param.reg))
+
+        parameters = [s.param for s in syscall_entry.params]
+        registers = [s.reg for s in syscall_entry.params]
+
+        info("Detected syscall {}".format(Color.colorify(syscall_entry.name, attrs=color)))
+        gef_print("    {}({})".format(syscall_entry.name, ', '.join(parameters)));
+
+        headers = [Color.colorify(x, attrs=color) for x in ["Parameter", "Register", "Value"]]
+        param_names = [re.split(' |\*', p)[-1] for p in parameters]
+        info("{:<28} {:<28} {}".format(*headers))
+        for i in range(len(parameters)):
```

Comment:
I think I'd prefer `for name, register, value in zip(param_names, registers, values):` or similar.